utilities/transactions/pessimistic_transaction.cc

```diff
@@ -166,11 +166,6 @@ template <typename TValue>
 inline Status WriteCommittedTxn::GetForUpdateImpl(
     const ReadOptions& read_options, ColumnFamilyHandle* column_family,
     const Slice& key, TValue* value, bool exclusive, const bool do_validate) {
-  if (read_options.io_activity != Env::IOActivity::kUnknown) {
-    return Status::InvalidArgument(
-        "Cannot call GetForUpdate with `ReadOptions::io_activity` != "
-        "`Env::IOActivity::kUnknown`");
-  }
```

Comment:
Cleanup: remove read options check on TXNs as I intended to add check to db.h level (e.g, TXN db) API only.